coreclient/src/clients/user_profile.rs

```diff
@@ -27,27 +28,31 @@ impl CoreUser {
         let mut notifier = self.store_notifier();
         let mut connection = self.pool().acquire().await?;
 
-        let user_profile_key = UserProfileKey::load_own(&mut *connection).await?;
+        let user_profile_key = UserProfileKey::random(self.user_name())?;
         let user_profile = IndexedUserProfile::new(
             user_profile_content.user_name,
             user_profile_key.index().clone(),
             user_profile_content.display_name,
             user_profile_content.profile_picture,
         );
 
-        // Phase 1: Store the user profile update in the database
+        // Phase 1: Store the user profile and the new key in the database
         user_profile.update(&mut *connection, &mut notifier).await?;
+        user_profile_key.store_own(&mut connection).await?;
+
+        let own_key = UserProfileKey::load_own(connection.as_mut()).await?;
```

Comment:
Isn't this operation redundant, because we have the key already?

Reply:
Right, sorry. That was a debugging artifact. I need to be more careful with those in the future.